### PR TITLE
Milestone: process data in models

### PR DESCRIPTION
**Pull request by Roseokpe**
In this milestone, I set up model for each tables for Users, Posts, Comments, and Likes.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] in these lines you should set the foreign key to the user.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Post < ApplicationRecord
+  # belongs_to :class_name: 'User', foreign_key: 'author_id'
+  # has_many :class_name: 'Comment', foreign_key: 'post_id'
+
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] you don't need to set the foreign_key in this line.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  # belongs_to :class_name: 'User', foreign_key: 'author_id'
+  # belongs_to :class_name: 'Post', foreign_key: 'post_id'
+  # has_many :post_like, through: :post, source: :author
+  # has_many :user_like, through: :user, source: :author
+
+  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (mikethreels) [inline app/models/like.rb]: - [ ] you don't need to set the foreign_key in this line.

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Comment < ApplicationRecord
+  # belongs_to :class_name: 'User', foreign_key: 'author_id'
+  # belongs_to :class_name: 'Post', foreign_key: 'post_id'
+  # has_many :post_comment, through: :post, source: :author
+  # has_many :user_comment, through: :user, source: :author
+  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (mikethreels) [inline app/models/comment.rb]: - [ ] you don't need to set the foreign_key in this line.

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
+
+  private
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (mikethreels) [inline app/models/comment.rb]: -[ ] update_likes_counter should be in the like model

File: app/models/post.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  private
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (mikethreels) [inline app/models/post.rb]: -[ ] update_likes_counter should be in the like model

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Rose, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 6 above)
Comment (Roseokpe) [reply app/models/post.rb]: Thank you @mikethreels

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Roseokpe , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**